### PR TITLE
Feature/autocomplete version 1

### PR DESCRIPTION
**Pull request by neubee**
Read through these changes and let me know if anything looks wrong or could use clarification. And try out this branch locally to see it working! I switched special rules effects and growth actions inputs to use autocomplete. You can go through and replace any input or textarea you want to with an AutoComplete element instead. Autocompletes look like this

```
<AutoComplete
  id={`elementId`}
  elementType="input"
  placeholder="placeholder text"
  bind:value={growthAction.effect} 
  {validAutoCompleteValues}
  // Optional props
  showListImmediately={true}
  endCharacters={["}", " "]
  startCharacter="{"
  />
```
id, elementType, placeholder, bind:value, and validAutoCompleteValues are all required.

elementType can be either `'input'` or `'textarea'`
validAutoCompleteValues needs to be an array with this structure
```
  [
    { label: "presence", value: "presence" },
    {
      label: "explorer",
      value: "explorer",
    },
    ...
  ];
```
label and value can be different. Label is what shows on the autocomplete list, value is what actually gets inserted into the input string.

showListImmediately should either not be included or set to true. It tells the autocomplete to start immediately without a start character.

startCharacter is used to start auto-completing when showListImmediately is not used and defaults to `'{'`

endCharacters is used to stop auto-completing and defaults to `[ '}', ' ' ]`

### CODE REVIEW

File: src/routes/spirit-board/growth.svelte [hunk 1]
```diff
@@ -221,12 +245,12 @@
...
+                    <AutoComplete
                       id={`growthSet${i}Group${j}Action${k}`}
-                      class="input"
-                      type="text"
-                      tabindex="1"
+                      elementType="input"
                       placeholder="Growth Action"
+                      showListImmediately={true}
+                      {validAutoCompleteValues}
                       on:blur={easyReport}
```
Comment (neubee) [inline src/routes/spirit-board/growth.svelte:254]: This on:blur event no longer runs, but I think we could pass a "do this on blur" function to AutoComplete and have it run there.

File: src/routes/spirit-board/growth.svelte [hunk 2]
```diff
@@ -221,12 +245,12 @@
               {#each growthGroup.growthActions as growthAction, k (growthAction.id)}
                 <div class="growth-action-container">
                   <div class="control">
-                    <input
+                    <AutoComplete
```
Comment (neubee) [inline src/routes/spirit-board/growth.svelte:248]: AutoComplete example for an input element.

**Comment by resonant-gamedesign** [comment]
That was a test anyway, could be removed

**Comment by neubee** [comment]
Limit option added. You can use it like this

`listLengthLimit={N}`

Where N is any number.

**Comment by neubee** [comment]
So maybe making the list scroll-able would be better than doing a limit, then there won't be hidden options. I think that could be accomplished with CSS only? Set a max height for the list and then use the overflow property to make it scroll. I'll need to try it out.

**Comment by resonant-gamedesign** [comment]
Yep, good call. scroll works much better

**Comment by resonant-gamedesign** [comment]
It seems like I can't set the tabindex of the autocomplete fields. how
would i do that?
Also, i think it would be good to have the lists for the icons like {dahan}
be shared between multiple sections. gotta do some import export stuff but
I don't quite know it .

**Comment by neubee** [comment]
> It seems like I can't set the tabindex of the autocomplete fields. how would i do that? 

You're saying you want to set the tabindex of the input elements, right? You can handle it just like properties that get passed into the component. 

- Add tabindex to the list of properties [here](https://github.com/neubee/spirit-island-builder/blob/9f96410ef1c693c2c3905422705a4c68ceee293f/src/routes/auto-complete/index.svelte#L4-L12) 
- Then set tabindex for the input elements using the property value [here](https://github.com/neubee/spirit-island-builder/blob/9f96410ef1c693c2c3905422705a4c68ceee293f/src/routes/auto-complete/index.svelte#L235) and [here](https://github.com/neubee/spirit-island-builder/blob/9f96410ef1c693c2c3905422705a4c68ceee293f/src/routes/auto-complete/index.svelte#L252)
- Then pass in tabindex just like all the other properties such as id, placeholder, startCharacter, etc.

> Also, i think it would be good to have the lists for the icons like {dahan} be shared between multiple sections. gotta do some import export stuff but I don't quite know it.

Assuming that our lists are all pretty static, it would be best to store them in another file similar to the `lib.js` file. It can be placed in the auto-complete directory and contain whatever lists we need as arrays. Look like this

```
export const innatePowersList = [...]
export const growthList = [...]
...
```

- Import that file into the auto-complete index.svelte file.
- Replace the `validAutoCompleteValues` property that currently is passed into the auto-complete component with the name of the list that should be used, such as `"growthList"`. Make sure to also rename `validAutoCompleteValues`.
- Then in the auto-complete component you can reference the list to use by doing `AutoCompleteLists[listName]` (replace AutoCompleteLists and validAutoCompleteValues with whatever names you want). So you would replace line 6 `export let validAutoCompleteValues;` with a line further down around 14 that looks like `const validAutoCompleteValues = AutoCompleteLists[listName]`.

**Comment by resonant-gamedesign** [comment]
can I apply a class to the autocomplete? I'm wanting to use the "is-small"
class.

**Pull request merged**